catalog/app/components/Markdown/Markdown.tsx

```diff
@@ -113,114 +112,86 @@ const highlight = (str: string, lang: string) => {
+    (tokens[idx] as CheckboxContentToken).checked ? '☑' : '☐'
+}
 
-/**
- * Create a plugin for remarkable that does custom processing of image tags.
- *
- * @param {Object} options
- * @param {bool} options.disable
- *   Don't show images, render them as they are in markdown contents (escaped).
- * @param {function} options.process
- *   Function that takes an image object ({ alt, src, title }) and returns a
- *   (possibly modified) image object.
- *
- * @returns {MarkdownPlugin}
- */
-const imageHandler =
-  ({ disable = false, process = R.identity }) =>
-  (md: Remarkable.Remarkable) => {
-    // eslint-disable-next-line no-param-reassign
-    md.renderer.rules.image = (tokens, idx) => {
-      const t = process(tokens[idx])
+type AttributeProcessor = (attr: string) => string
 
-      if (disable) {
-        const alt = t.alt ? escape(t.alt) : ''
-        const src = escape(t.src)
-        const title = t.title ? ` "${escape(t.title)}"` : ''
-        return `<span>![${alt}](${src}${title})</span>`
-      }
+function handleImage(process: AttributeProcessor, element: Element): Element {
+  const attributeValue = element.getAttribute('src')
+  if (!attributeValue) return element
+  const result = process(attributeValue)
+  element.setAttribute('src', result)
 
-      const src = (Remarkable as unknown as RemarkableWithUtils).utils.escapeHtml(t.src)
-      const alt = t.alt
-        ? escape((Remarkable as unknown as RemarkableWithUtils).utils.unescapeMd(t.alt))
-        : ''
-      const title = t.title ? ` title="${escape(t.title)}"` : ''
-      return `<img src="${src}" alt="${alt}"${title} />`
-    }
+  const alt = element.getAttribute('alt')
+  if (alt) {
+    element.setAttribute('alt', unescapeMd(alt))
+  }
+  const title = element.getAttribute('title')
+  if (title) {
+    element.setAttribute('title', title)
   }
 
-/**
- * Create a plugin for remarkable that does custom processing of links.
- *
- * @param {Object} options
- * @param {bool} options.nofollow
- *   Add rel="nofollow" attribute if true (default).
- * @param {function} options.process
- *   Function that takes a link object ({ href, title }) and returns a
- *   (possibly modified) link object.
- *
- * @returns {MarkdownPlugin}
- */
-const linkHandler =
-  ({ nofollow = true, process = R.identity }) =>
-  (md: Remarkable.Remarkable) => {
-    // eslint-disable-next-line no-param-reassign
-    md.renderer.rules.link_open = (tokens, idx) => {
-      const t = process(tokens[idx])
-      const title = t.title ? ` title="${escape(t.title)}"` : ''
-      const rel = nofollow ? ' rel="nofollow"' : ''
-      return `<a href="${(Remarkable as unknown as RemarkableWithUtils).utils.escapeHtml(
-        t.href,
-      )}"${rel}${title}>`
-    }
+  return element
+}
+
+function handleLink(process: AttributeProcessor, element: HTMLElement): Element {
+  const attributeValue = element.getAttribute('href')
+  if (!attributeValue) return element
```

Comment:
if we short-circuit here, `setAttribute` calls below won't be executed. is this intentional?

Reply:
Yes, it was. But I found that we process `href=""` (as relative path to the root of the package), so I changed this condition to `if (typeof attributeValue !== 'string')`